easybuild/easyblocks/o/orca.py

```diff
@@ -0,0 +1,142 @@
+# the Flemish Supercomputer Centre (VSC) (https://www.vscentrum.be),
+# Flemish Research Foundation (FWO) (http://www.fwo.be/en)
+# and the Department of Economy, Science and Innovation (EWI) (http://www.ewi-vlaanderen.be/en).
+#
+# https://github.com/easybuilders/easybuild
+#
+# EasyBuild is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation v2.
+#
+# EasyBuild is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with EasyBuild.  If not, see <http://www.gnu.org/licenses/>.
+##
+"""
+EasyBuild support for ORCA, implemented as an easyblock
+
+@author: Alex Domingo (Vrije Universiteit Brussel)
+"""
+import glob
+import os
+
+from distutils.version import LooseVersion
+from easybuild.easyblocks.generic.makecp import MakeCp
+from easybuild.easyblocks.generic.packedbinary import PackedBinary
+from easybuild.framework.easyconfig import CUSTOM
+from easybuild.tools.build_log import EasyBuildError
+from easybuild.tools.filetools import write_file
+from easybuild.tools.py2vs3 import string_type
+
+
+class EB_ORCA(PackedBinary, MakeCp):
+    """Support for installing ORCA."""
+
+    @staticmethod
+    def extra_options(extra_vars=None):
+        """Extra easyconfig parameters for ORCA."""
+        extra_vars = MakeCp.extra_options()
+        extra_vars.update(PackedBinary.extra_options())
+
+        # files_to_copy is not mandatory here, since we set it by default in install_step
+        extra_vars['files_to_copy'][2] = CUSTOM
+
+        return extra_vars
+
+    def install_step(self):
+        """Install ORCA with MakeCp easyblock"""
+
+        if not self.cfg['files_to_copy']:
+            # Put installation files in standard locations
+            files_to_copy = [
+                (['auto*', 'orca*', 'otool*'], 'bin'),
+                (['*.pdf'], 'share'),
+            ]
+            # Version 5 extra files
+            if LooseVersion(self.version) >= LooseVersion('5.0.0'):
+                compoundmethods = (['ORCACompoundMethods'], 'bin')
+                files_to_copy.append(compoundmethods)
+            # Shared builts have additional libraries
+            libs_to_copy = (['liborca*'], 'lib')
+            if all([glob.glob(p) for p in libs_to_copy[0]]):
+                files_to_copy.append(libs_to_copy)
+
+            self.cfg['files_to_copy'] = files_to_copy
+
+        MakeCp.install_step(self)
+
+    def sanity_check_step(self):
+        """Custom sanity check for ORCA"""
+
+        custom_paths = {'files': [], 'dirs': []}
+
+        # Convert 'files_to_copy' to list of files in build directory
+        for spec in self.cfg['files_to_copy']:
```

Comment:
When using `--module-only` the `files_to_copy` configuration option is not going to exist, this will lead to a failure like:
```
#### Overview of tested easyconfigs (in order)
 * **FAIL (unhandled exception: 'NoneType' object is not iterable)Traceback (most recent call last):
  File "/users/ocaisa/EasyBuild_Git/easybuild-framework/easybuild/main.py", line 118, in build_and_install_software
    (ec_res['success'], app_log, err) = build_and_install_one(ec, init_env)
  File "/users/ocaisa/EasyBuild_Git/easybuild-framework/easybuild/framework/easyblock.py", line 3691, in build_and_install_one
    result = app.run_all_steps(run_test_cases=run_test_cases)
  File "/users/ocaisa/EasyBuild_Git/easybuild-framework/easybuild/framework/easyblock.py", line 3582, in run_all_steps
    self.run_step(step_name, step_methods)
  File "/users/ocaisa/EasyBuild_Git/easybuild-framework/easybuild/framework/easyblock.py", line 3435, in run_step
    step_method(self)()
  File "/tmp/eb-2z1tnfw7/included-easyblocks-o5uxtum4/easybuild/easyblocks/orca.py", line 84, in sanity_check_step
    for spec in self.cfg['files_to_copy']:
TypeError: 'NoneType' object is not iterable
** _ORCA-4.2.1-gompi-2019b.eb_ 
```
Is there maybe a minimal list that can be included here in the case that can be included as a fallback for this case (and then we are primarily relying on the sanity check command)

Reply:
very good point, I didn't think about that case. Fixed in https://github.com/easybuilders/easybuild-easyblocks/pull/2504/commits/1903821afd115bee2b319e91173843b999a70bfb and https://github.com/easybuilders/easybuild-easyblocks/pull/2504/commits/5fb552b6197a18e8208c6bf35299dc19d2e349f7